src/lib/webSocketClient.ts

```diff
@@ -0,0 +1,141 @@
+import { v4 as genUuid } from 'uuid';
+import camelcaseKeysDeep from 'camelcase-keys-deep';
+import { IMiddlewareWebSocketSubscriptionMessage, WebSocketChannelName } from '../types';
+import {
+  WEB_SOCKET_CHANNELS,
+  WEB_SOCKET_SUBSCRIBE,
+  WEB_SOCKET_UN_SUBSCRIBE,
+  handleUnknownError,
+} from '../popup/utils';
+
+let wsClient: WebSocket;
+let isWsConnected: boolean;
+
+const subscribersQueue: IMiddlewareWebSocketSubscriptionMessage[] = [];
+// eslint-disable-next-line no-unused-vars
+const subscribers:Record<WebSocketChannelName, Record<string, (payload: any) => void>> = {
+  [WEB_SOCKET_CHANNELS.Transactions]: {},
+  [WEB_SOCKET_CHANNELS.MicroBlocks]: {},
+  [WEB_SOCKET_CHANNELS.KeyBlocks]: {},
+  [WEB_SOCKET_CHANNELS.Object]: {},
+};
+
+export default class WebSocketClient {
```

Comment:
This class has only static methods so it is a static class. Static classes should not operate on any state, but actually there is a state: `wsClient` and `isWsConnected`. This is really bad pattern to mix the concept of regular classes (which should encapsulate state and operations) with the static/singleton. So please use encapsulated state or convert the class into singleton.